src/Presentation/Nop.Web/Areas/Admin/Controllers/ProductController.cs

```diff
@@ -739,6 +739,48 @@ protected virtual async Task GenerateAttributeCombinationsAsync(Product product,
 
         #region Product list / create / edit / delete
 
+        public virtual async Task<IActionResult> ViewById(int id)
+        {
+            if (!await _permissionService.AuthorizeAsync(StandardPermissionProvider.ManageProducts))
+                return AccessDeniedView();
+
+            //try to get a product with the specified id
+            var product = await _productService.GetProductByIdAsync(id);
+            if (product == null || product.Deleted)
+                return RedirectToAction("List");
+
+            //a vendor should have access only to his products
+            var currentVendor = await _workContext.GetCurrentVendorAsync();
+            if (currentVendor != null && product.VendorId != currentVendor.Id)
+                return RedirectToAction("List");
+
+            //prepare model
+            var model = await _productModelFactory.PrepareProductModelAsync(null, product);
+
+            //try to get a product picture with the specified id
+            var productPicture = await _productService.GetProductPictureByIdAsync(model.Id)
```

Comment:
how about the case when product will not have image at all, will this throw an exception?, because by design i can have a product without image